src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,328 @@ private static string ToHtmlCase(string name)
+                {
+                    Next();
+                }
+                var nameEndIndex = _index;
+
+                PassOptionalWhitespace();
+
+                var attributeNameLength = 0;
+                if (At('='))
+                {
+                    attributeNameLength = nameEndIndex - nameStartIndex;
+
+                    // If this proves true it means there was 0 whitespace between the attribute name and the '='.
+                    // Therefore, the operator would be baked into the attribute name by default.
+                    var potentialOperator = _requiredAttributes[_index - 1];
+                    if (TagHelperRequiredAttributeDescriptor.IsSupportedCSSValueOperator(potentialOperator))
+                    {
+                        attributeNameLength--;
+                    }
+                }
+                else if (At(']'))
+                {
+                    attributeNameLength = nameEndIndex - nameStartIndex;
+                }
+                else if (NextIs('=') && TagHelperRequiredAttributeDescriptor.IsSupportedCSSValueOperator(Current))
+                {
+                    // Move past selector
+                    Next();
+                    attributeNameLength = nameEndIndex - nameStartIndex;
+                }
+                else
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Create error for unknown character not matching a square braces.", 0);
+                    return null;
+                }
+
+                var attributeName = _requiredAttributes.Substring(nameStartIndex, attributeNameLength);
+
+                return attributeName;
+            }
+
+            private char ParseCSSValueOperator(ErrorSink errorSink)
+            {
+                Debug.Assert(Current == '=');
+
+                // Move past the '='
+                Next();
+                var potentialOperator = _requiredAttributes[_index - 2];
+                if (TagHelperRequiredAttributeDescriptor.IsSupportedCSSValueOperator(potentialOperator))
+                {
+                    return potentialOperator;
+                }
+
+                return '=';
+            }
+
+            private string ParseCSSValue(ErrorSink errorSink)
+            {
+                PassOptionalWhitespace();
+
+                int valueStart, valueEnd;
+                if (At('\'') || At('"'))
+                {
+                    var quote = Current;
+
+                    // Move past the quote
+                    Next();
+
+                    valueStart = _index;
+                    while (!At(quote))
+                    {
+                        if (AtEnd)
+                        {
+                            errorSink.OnError(SourceLocation.Zero, "TODO: Mismatching quotes", 0);
+                            return null;
+                        }
+
+                        Next();
+                    }
+                    valueEnd = _index;
```

Comment:
Remove the `while`, `valueEnd = _requiredAttributes.IndexOf(quote, nameStartIndex)` and add an error if `valueEnd == -1`.